controller/apps.go

```diff
@@ -0,0 +1,443 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/auth/authservice"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/auth"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	witerrors "github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/kubernetes"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AppsController implements the apps resource.
+type AppsController struct {
+	*goa.Controller
+	Config *configuration.Registry
+}
+
+// NewAppsController creates a apps controller.
+func NewAppsController(service *goa.Service, config *configuration.Registry) *AppsController {
+	return &AppsController{
+		Controller: service.NewController("AppsController"),
+		Config:     config,
+	}
+}
+
+func tostring(item interface{}) string {
+	bytes, _ := json.MarshalIndent(item, "", "  ")
+	return string(bytes)
+}
+
+func (c *AppsController) getAndCheckOSIOClient(ctx context.Context) *OSIOClient {
+
+	// TODO - grab the WIT URL from the incoming request if possible
```

Comment:
that would be found in the `ctx.RequestURI` of the endpoint function

Reply:
I've fixed this, but I'm thinking I should force the host to localhost - in some configurations using a network address other than loopback actually goes out to the network. (~6% on my laptop).
As ultimately this may not run on the same system as WIT, I'll leave that for later.